src/main/java/com/es/phoneshop/model/exceptions/OutOfStockException.java

```diff
@@ -0,0 +1,28 @@
+package com.es.phoneshop.model.exceptions;
+
+import com.es.phoneshop.model.product.Product;
+
+public class OutOfStockException extends Exception{
+    private Product product;
```

Comment:
You shouldn't store any business logic-related fields into the exception. Generally it is sufficient to have only message and root cause. In some specific situation it's also allowed to introduce new exception-specific field, e.g. errorCode, status, etc.